meldebot/mel/gif.py

```diff
@@ -15,13 +15,32 @@
 from requests import get as http_get
 
 # Self imports
-from meldebot.mel.conf import get_giphy_api_key, get_debug_enabled
+from meldebot.mel.conf import get_giphy_api_key, get_debug_enabled, get_tenor_api_key
 from meldebot.mel.utils import send_typing_action, remove_command_message
 
 logger = getLogger(__name__)
 
 
-def get_gif_url(params):
+def get_gif_url(params, provider=None):
+    if not provider:
+        provider = get_random_gif_provider()
+    return provider(params)
+
+
+def get_random_gif_provider():
+    return get_gif_provider(choice(list(GIF_PROVIDERS.keys())))
+
+
+def get_gif_provider(provider_name):
+    method_caller = GIF_PROVIDERS.get(provider_name)
+    if not method_caller:
+        logger.critical("NO PROIDER DEFINED FOR {0}".format(provider_name))
+        exit(-1)  # exit or raise Exception ?
```

Comment:
Make it raise an exception with the same text, that way it will be logged as it should.
Also we run this with python 3.7, so you can use f-strings 👍 

```suggestion
    if not method_caller:
        raise Exception(f"No provider defined with name: {provider_name}")
```